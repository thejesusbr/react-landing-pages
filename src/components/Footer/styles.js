import styled, { css } from 'styled-components';
import { SimpleText as TextComponent } from '../TextComponent/styles';
import { MyStyle as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    background: ${theme.colors.lightColor};
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.secondaryColor};
    a {
      color: inherit;
      text-decoration: none;
    }
    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
