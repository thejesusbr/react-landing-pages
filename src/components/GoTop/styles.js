import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.lightColor};
    border: 1px solid ${theme.colors.lightColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: ${theme.font.sizes.xlarge};
    height: ${theme.font.sizes.xlarge};
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;

    & svg {
      width: 2.5rem;
      height: 2.5rem;
    }
  `}
`;
