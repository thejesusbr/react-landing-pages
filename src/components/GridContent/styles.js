import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 80%;
    text-align: center;
    margin: auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`;
