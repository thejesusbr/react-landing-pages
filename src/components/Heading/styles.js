import styled, { css } from 'styled-components';

const titleSizes = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.normal};
  `,
  normal: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  large: (theme) => css`
    font-size: ${theme.font.sizes.huge};
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaQueryFont(theme)};
  `,
};

const mediaQueryFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.large};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, darkBG, size, uppercase }) => css`
    color: ${darkBG ? theme.colors.white : theme.colors.primaryColor};
    ${titleSizes[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
