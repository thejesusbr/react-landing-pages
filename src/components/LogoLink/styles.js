import styled, { css } from 'styled-components';

export const MyStyle = styled.a`
  ${({ theme }) => css`
    text-decoration: none;
    color: inherit;
    align-items: center;

    img {
      max-height: 3rem;
    }
  `}
`;
