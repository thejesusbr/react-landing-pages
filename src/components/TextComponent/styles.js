import styled, { css } from 'styled-components';

export const SimpleText = styled.p`
  ${({ theme }) =>
    css`
      font-size: ${theme.font.sizes.normal};
    `}
`;
