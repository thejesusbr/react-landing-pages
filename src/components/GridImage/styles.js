import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { SimpleText as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextContainer} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    height: auto;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
