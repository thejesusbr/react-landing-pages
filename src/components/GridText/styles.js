import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { SimpleText as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextContainer} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.huge};
      top: -1rem;
      left: -${theme.spacings.xlarge};
      transform: rotate(5deg);
    }

    @media ${theme.media.lteMedium} {
      ${HeadingContainer}::before {
        font-size: ${theme.font.sizes.xlarge};
        top: -0.75rem;
        left: -${theme.spacings.large};
      }
    }
  `}
`;
