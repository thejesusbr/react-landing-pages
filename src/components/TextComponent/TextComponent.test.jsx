import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { TextComponent } from '.';

describe('<TextComponent />', () => {
  it('should render a text', () => {
    renderTheme(<TextComponent>Text</TextComponent>);
    const texto = screen.getByText('Text');
    expect(texto).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { container } = renderTheme(<TextComponent>Text</TextComponent>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
