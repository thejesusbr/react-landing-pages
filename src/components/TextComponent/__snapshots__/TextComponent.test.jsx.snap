// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent /> should match the snapshot 1`] = `
.c0 {
  font-size: 2.4rem;
}

<p
  class="c0"
>
  Text
</p>
`;
